paper-icon-button.html

```diff
@@ -49,106 +49,161 @@
 
 The opacity of the ripple is not customizable via CSS.
 
-Accessibility
--------------
-
-The button is accessible by default if you use the `icon` property. By default, the
-`aria-label` attribute will be set to the `icon` property. If you use a custom icon,
-you should ensure that the `aria-label` attribute is set.
-
-    <paper-icon-button src="star.png" aria-label="star"></paper-icon-button>
-
 @element paper-icon-button
-@extends paper-button-base
 @homepage github.io
 -->
 
-<link href="../polymer/polymer.html" rel="import">
-<link href="../core-icon/core-icon.html" rel="import">
-<link href="../core-icons/core-icons.html" rel="import">
-<link href="../paper-button/paper-button-base.html" rel="import">
-<link href="../paper-ripple/paper-ripple.html" rel="import">
-
-<polymer-element name="paper-icon-button" extends="paper-button-base" attributes="src icon" role="button">
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../core-icon/core-icon.html">
+<link rel="import" href="../core-focusable/core-focused-state.html">
+<link rel="import" href="../core-focusable/core-disabled-state.html">
+<link rel="import" href="../core-focusable/core-active-state.html">
+<link rel="import" href="../paper-button/paper-button-state.html">
+<link rel="import" href="../paper-ripple/paper-ripple.html">
+<link rel="import" href="../paper-card/paper-card.html">
+<link rel="import" href="../layout/layout.html">
+
+<dom-module id="paper-icon-button">
+  <!--<link rel="import" type="css" href="../paper-button/paper-button.css">-->
+  <style>
+    /* TODO(cdata): Currently we are targeting :host([role]) just so that
+       we can bump the specificity slightly. This should be updated when
+       polymer/polymer#1349 is resolved */
+    :host {
+      display: inline-block;
+      position: relative;
+      padding: 8px;
+      outline: none;
+      -webkit-user-select: none;
+      -moz-user-select: none;
+      -ms-user-select: none;
+      user-select: none;
+      cursor: pointer;
+      z-index: 0;
+    }
 
+    :host([disabled]) {
+      color: #c9c9c9;
+      pointer-events: none;
+      cursor: auto;
+    }
+  </style>
   <template>
+    <template is="core-disabled-state" disabled="{{disabled}}"></template>
+    <template is="core-focused-state" focused="{{focused}}" disabled="[[disabled]]"></template>
+    <template is="core-active-state" active="{{active}}" disabled="[[disabled]]" toggle="[[toggle]]"></template>
+    <template is="paper-button-state"
+      elevation="{{elevation}}" toggle="{{toggle}}" raised="true"
+      active="[[active]]" focused="[[focused]]" disabled="[[disabled]]"></template>
+
+    <paper-ripple></paper-ripple>
```

Comment:
Will double check. I may only be able to fix so much, though, since I handed paper-ripple work off to Scott H.